src/OSPSuite.UI/Services/DataImporter.cs

```diff
@@ -121,7 +128,8 @@ DataImporterSettings dataImporterSettings
                importerPresenter.LoadConfiguration(configuration);
                using (var importerModalPresenter = _container.Resolve<IModalImporterPresenter>())
                {
-                  return importerModalPresenter.ImportDataSets(importerPresenter, metaDataCategories, columnInfos, dataImporterSettings).DataRepositories;
+                  return importerModalPresenter.ImportDataSets(importerPresenter, metaDataCategories, columnInfos, dataImporterSettings)
```

Comment:
here and in other functions: why are `IModalImporterPresenter`, `IImporterPresenter`, ... created via `_container.Resolve` and not via constructor injection?

Reply:
They are modal presenters and should be disposed explicitly. This is correct but one should use _applicationController instead ;. Not sure why we need the ref to _container at all

Reply:
substituted with applicationController.Start<>(). container remove since this was the only place it was used